src/tir/schedule/primitive/compute_at.cc

```diff
@@ -160,7 +174,57 @@ int FindInsertionPoint(
   // The valid indices are: (last_producer_position, first_consumer_position]
   ICHECK(split.last_producer_position < split.first_consumer_position);
   // Step 4. Return the last valid insertion point
-  return split.first_consumer_position;
+  int insert_position = split.first_consumer_position;
+  if (index == -1) {
+    return insert_position;
+  } else if (index == -2) {
+    if (require_all_producers_visited) {
+      if (split.last_producer_position >= 0) {
+        insert_position = split.last_producer_position + 1;
+      } else {
+        insert_position = split.first_consumer_position;
+      }
```

Comment:
This line does not make sense to me. `split.last_producer_position` should always be `>=-1`, so `split.last_producer_position + 1`, which is `>=0` is always a valid position for `index == -2`

lmk if I make some mistakes.

Reply:
when  split.last_producer_position==-1, we still hope the insert_position is split.first_consumer_position(the last block) instead of 0.